include/usb_cam/formats/mjpeg.hpp

```diff
@@ -59,10 +59,96 @@ namespace usb_cam
 namespace formats
 {
 
+class RAW_MJPEG : public pixel_format_base
+{
+public:
+  RAW_MJPEG(const AVPixelFormat & avDeviceFormat)
+  : pixel_format_base(
+      "raw_mjpeg",
+      V4L2_PIX_FMT_MJPEG,
+      usb_cam::constants::YUV422,
+      2,
+      8,
+      false)
+  {
+    switch(avDeviceFormat)
```

Comment:
@boitumeloruf this switch should be moved to its own function. input is the avDeviceFormat, output would be the corresponding ROS format. Should probably split it into two separate functions - one for the ROS format, the other for the channels.

Something like `get_ros_format_from_av_format(const AVPixelFormat & avDeviceFormat)` and then same thing for the channels.